### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by ArtemLeo**

### CODE REVIEW

File: city/routers.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+async def read_cities(
+        skip: int = 0,
+        limit: int = 10,
+        db: AsyncSession = Depends(get_db)
+) -> List[schemas.City]:
+    cities = await crud.get_cities(db, skip=skip, limit=limit)
+    return cities
+
+
+@router.get("/{city_id}", response_model=schemas.City)
```
Comment (vsmutok) [inline city/routers.py]:
```suggestion
@router.get("/{city_id}/", response_model=schemas.City)

```

File: city/routers.py [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+        city_id: int,
+        db: AsyncSession = Depends(get_db)
+) -> schemas.City:
+    city = await crud.get_city(db, city_id=city_id)
+    if city is None:
+        raise HTTPException(status_code=404, detail="City not found")
+    return city
+
+
+@router.delete("/{city_id}")
```
Comment (vsmutok) [inline city/routers.py]:
```suggestion
@router.delete("/{city_id}/")

```

File: city/routers.py [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+async def delete_city(city_id: int, db: AsyncSession = Depends(get_db)) -> dict:
+    await crud.delete_city(db, city_id)
+    return {"ok": True}
+
+
+@router.put("/{city_id}", response_model=schemas.City)
```
Comment (vsmutok) [inline city/routers.py]:
```suggestion
@router.put("/{city_id}/", response_model=schemas.City)

```

File: temperature/routers.py [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+        async with db_session_factory() as new_db_session:
+            await crud.create_temperature(new_db_session, temperature)
+    except httpx.HTTPStatusError as exc:
+        print(f"Error response {exc.response.status_code} "
+              f"while requesting {exc.request.url!r}.")
+    except Exception as e:
+        print(f"An error occurred: {str(e)}")
+
+
+@router.post("/update")
```
Comment (vsmutok) [inline temperature/routers.py]:
```suggestion
@router.post("/update/")

```

File: temperature/routers.py [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+async def read_temperatures(
+        db: AsyncSession = Depends(get_db)
+) -> List[schemas.Temperature]:
+    temperatures = await crud.get_temperatures(db)
+    if not temperatures:
+        raise HTTPException(status_code=404, detail="No temperatures found")
+    return temperatures
+
+
+@router.get("/{city_id}", response_model=list[schemas.Temperature])
```
Comment (vsmutok) [inline temperature/routers.py]:
```suggestion
@router.get("/{city_id}/", response_model=list[schemas.Temperature])

```

**Review by vsmutok — CHANGES_REQUESTED** [review]

**Review by pavlopro — CHANGES_REQUESTED** [review]
Remove .idea and add it to gitignore

**Pull request closed without merging**